### PR TITLE
Move plotting to submodule

### PR DESCRIPTION
**Pull request by nvaytet**
The plotting files were starting to get large, and difficult to maintain.
This PR moves them to a `plot` submodule and split the functions into a larger number of smaller files.

Additional notes:

- the `sc.plot_config` is now `sc.plot.config`
- the `sc.plot` function is now `sc.plot.plot`, so importing it as `from scipp.plot import plot` has been introduced in the notebooks
- introduced a static image renderer for `plotly` plots, that actually show up in the docs
- ~~I therefore propose to remove the `matplotlib` backend, thus avoiding having to maintain two different backends~~
- the available backends are now called `'interactive'` (for Jupyter notebook plots) and `'static'` (for png exports), and `'matplotlib'` for returning a dict of mpl objects
- the rasterization threshold for large images can now be changed inside the `config`: `config.rasterize_threshold = 100000`

**New feature**:
You can now use the `'matplotlib'` backend to get a dict of mpl objects for maximum control over your plots. This is best illustrated with a demo:

Screenshot at 2019-10-16 16-08-46

### CODE REVIEW

**Comment by nvaytet** [comment]
> Great improvements!
> 
> > ```
> > * I therefore propose to remove the `matplotlib` backend, thus avoiding having to maintain two different backends
> > ```
> 
> Is it worth keeping around for expert who want to create/customize their own matplotlib plots? That is, have the basics (1-D plot, histogram plot, 2-D plot) but not support all the advanced features (rasterization, sliders, ...).

hmmm I would say if an expert really wants to customize plots, say doing something with different panels
Screenshot at 2019-10-16 11-25-47
then a utility that automatically tries to something for you won't be super useful. For instance, you might want to have latex code inside the axes labels or titles, and you can't access that in the simple `matplotlib` utility we have. We would have to make it very customizable... Unless we can return the figure/axes objects which can then be further customized by changing their attibutes?

**Comment by SimonHeybrock** [comment]
You are right I guess. Maybe I should reformulate: Is there anything we can do to make creating such custom plots easy and convenient, e.g., provide a function that returns a data array as a dict, that does all the scipp-specific pre-processing (not what what would be needed, I imagine transposing, rendering dim labels as strings, rendering units as strings, ...). You probably have all this embedded in the plotting code somewhere, and maybe it is worth making it accessible?

File: docs/tutorials/multi-d-datasets.ipynb [hunk 1]
```diff
@@ -19,7 +19,8 @@
    "source": [
     "import numpy as np\n",
     "import scipp as sc\n",
-    "from scipp import Dim"
+    "from scipp import Dim\n",
+    "from scipp.plot import plot"
```
[the remark this answers is not in the document]
Comment (nvaytet) [reply docs/tutorials/multi-d-datasets.ipynb]: It has been added in most places I think:
https://github.com/scipp/scipp/pull/614/files#diff-5ee3c76144aca390fdd19f8d67a31be4R250
https://github.com/scipp/scipp/pull/614/files#diff-209ab6a9307624a1b3e8ae4e7e234367R23
https://github.com/scipp/scipp/pull/614/files#diff-3b1097395f1d77926d83cb4c3d99787bR20

File: docs/getting-started/quick-start.ipynb [hunk 2]
```diff
@@ -155,7 +155,7 @@
    },
    "outputs": [],
    "source": [
-    "sc.plot(dataset, backend='matplotlib')"
+    "sc.plot(dataset)"
```
Comment (SimonHeybrock) [inline docs/getting-started/quick-start.ipynb]: @nvaytet Not here?

File: docs/tutorials/multi-d-datasets.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/tutorials/multi-d-datasets.ipynb]: See other comment. Did a local build of the docs succeed?

File: docs/getting-started/quick-start.ipynb (hunk 2 above)
Comment (nvaytet) [reply docs/getting-started/quick-start.ipynb]: correct, that one is broken!
i think we need a docs test also on travis..

**Comment by nvaytet** [comment]
It is also a shame I couldn't get the slider/buttons widgets to show up on the docs pages. They claim it should be possible, and they indeed appear on their own doc pages: https://ipywidgets.readthedocs.io/en/latest/embedding.html#using-the-nbsphinx-project
But i could not get this to work locally. In addition, when you hit “Save Notebook Widget State”, the size of the notebook becomes ridiculous, even for just 4 buttons (by ridiculous i mean 50,000+ lines of code)... So i think for now it's fine to just have the images.

**Comment by nvaytet** [comment]
> You are right I guess. Maybe I should reformulate: Is there anything we can do to make creating such custom plots easy and convenient, e.g., provide a function that returns a data array as a dict, that does all the scipp-specific pre-processing (not sure what would be needed, I imagine transposing, rendering dim labels as strings, rendering units as strings, ...). You probably have all this embedded in the plotting code somewhere, and maybe it is worth making it accessible?

Yes, maybe we could do that, as people will use `matplotlib` a lot for scientific articles. As you suggest, we could return a dict that contains all the objects in the plot, the `fig`, the `axes`, and all the plot elements (lines, images, contours etc...)
Good idea!

**Pull request merged**